ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -235,11 +237,19 @@ storeLedgerStateAt slotNo (AnalysisEnv { db, registry, initLedger, cfg, ledgerDb
   Auxiliary: processing all blocks in the DB
 -------------------------------------------------------------------------------}
 
+data Limit = Limit Int | Unlimited
+
+decreaseLimit :: Limit -> Limit
+decreaseLimit Unlimited = Unlimited
+decreaseLimit (Limit 0) = Limit 0
```

Comment:
`0 -> 0` is a little unexpected. One way to avoid it:

```
decreaseLimit :: Limit -> Maybe Limit
...
decreaseLimit (Limit 0) = Nothing
```

And then use `case decreaseLimit lt of ...` instead of `case lt of ...` in the `go` loops.

Reply:
I don't like this idea. The whole point to have a function was to call it without pattern matching on the Limit. Now I will just change the Limit to Maybe, and will have to pattern match anyway. If that's the case then I would rather prefer pattern matching on the Limit directly and remove the `decreaseLimit` function.

I still think `decreaseLimit` makes sense, but maybe it should have been a private function, defined only in that one single place where it is being used.

Reply:
Ok, after a call, I get what you meant.